src/takedown.js

```diff
@@ -0,0 +1,85 @@
+import { getRepliesInBatch, getUserReplies } from './util/graphql.js';
+import { getSpamList } from './getSpamList.js';
+import { getAllPRs, createPullRequest } from './githubPR.js';
+import subTime from 'date-fns/sub';
+
+const knownUsers = new Set();
+
+async function initKnownUsers() {
+  const list = await getAllPRs();
+  list.forEach((pr) => {
+    // get userid from lastWord of pr title
+    const title = pr.title;
+    if (title.startsWith('Takedown spam user')) {
+      const userid = pr.title.slice(pr.title.lastIndexOf(' ') + 1);
+      // console.log('userid: ', userid);
+      if (!knownUsers.has(userid)) knownUsers.add(userid);
+    }
+  });
+}
+
+async function getSpamRepliesFromDate(date) {
+  try {
+    let allSpamReplies = [];
+
+    // get all replies from date to now
+    for await (const replies of getRepliesInBatch(
+      date.toISOString(),
+      new Date().toISOString()
+    )) {
+      // filter out known users
+      const filteredReplies = replies.filter(
+        (reply) => !knownUsers.has(reply.user.id)
+      );
+
+      const list = await getSpamList(filteredReplies);
+
+      // add new spam users to knownUsers, we only need to create PR for them once
+      list.forEach((reply) => {
+        knownUsers.add(reply.user.id);
+      });
+
+      allSpamReplies = allSpamReplies.concat(list);
+    }
+
+    // deduplicate
+    const replyMap = new Map();
+    allSpamReplies.forEach((reply) => {
+      if (!replyMap.get(reply.user.id)) {
+        replyMap.set(reply.user.id, reply);
+      }
+    });
+    const uniqueSpamReplies = Array.from(replyMap.values());
```

Comment:
Judging from the next `Promise.all()` step, what we want here is a list of spam users, rather than "unique spam replies". Reply itself is not relevant here, as we will get list of their replies in L57.

Reply:
There may be a case that user posts normal (latest)reply after posting spam. If I use uniqueUsers in this step then choosing latest userReplyHistory as `trimmedContent` in `createPullRequest`, the `trimmedContent` may be a normal reply.
But I doubt there are very few cases like this.